crates/oxc_minifier/src/ast_passes/peephole_minimize_conditions.rs

```diff
@@ -383,6 +383,76 @@ impl<'a> PeepholeMinimizeConditions {
             }
         }
 
+        let in_boolean_context = matches!(
+            ctx.parent(),
+            Ancestor::IfStatementTest(_)
+                | Ancestor::WhileStatementTest(_)
+                | Ancestor::DoWhileStatementTest(_)
+                | Ancestor::ExpressionStatementExpression(_)
+        );
```

Comment:
this is a very pessimistic implementation.
we can continue up the tree if we are in a logical expression/unary expression

e.g. `if ((foo ? bar : false) && que` would be changed to `if ((!!foo && bar) && que)` but we can drop the `!!` here since it's a boolean ctx